### PR TITLE
ceph: support crypt type device in OSD on PVC

### PR DESCRIPTION
**Pull request by satoru-takeuchi**
**Description of your changes:**

Support crypt type device in OSD on PVC.

**Which issue is resolved by this Pull Request:**

Nothing

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://rook.io/docs/rook/master/development-flow.html#commit-structure).
- [ ] **Skip Tests for Docs**: Add the flag for skipping the build if this is only a documentation change. See [here](https://github.com/rook/rook/blob/master/INSTALL.md#skip-ci) for the flag.
- [ ] **Skip Unrelated Tests**: Add a flag to run tests for a specific storage provider. See [test options](https://github.com/rook/rook/blob/master/INSTALL.md#test-storage-provider).
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/master/development-flow.html#submitting-a-pull-request)
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.

[test ceph]

### CODE REVIEW

**Comment by satoru-takeuchi** [comment]
I confirmed this PR works well. However, adding a test is necessary to be merged.

**Comment by leseb** [comment]
@satoru-takeuchi any plans to move forward with this? Thanks.

**Comment by satoru-takeuchi** [comment]
@leseb I want to support it in v1.4. I'd like to add this feature after merging the following PR.

https://github.com/rook/rook/pull/5741

It's because OSD on PVC can only be tested after merging the above-mentioned PR. I'm glad if you take a look at this PR.

If it's not necessary to add CI for now, I'll revise this PR as soon as possible.

**Comment by travisn** [comment]
> @leseb I want to support it in v1.4. I'd like to add this feature after merging the following PR.
> 
> #5741
> 
> It's because OSD on PVC can only be tested after merging the above-mentioned PR. I'm glad if you take a look at this PR.
> 
> If it's not necessary to add CI for now, I'll revise this PR as soon as possible.

@satoru-takeuchi Ideally we would have the integration tests with the feature, but it's not a strict requirement. We could merge this first, based on unit tests and manual testing of the feature. Then separately we could finalize the integration tests. Have you already manually tested the crypt devices for this PR?

**Comment by satoru-takeuchi** [comment]
> @satoru-takeuchi Ideally we would have the integration tests with the feature, but it's not a strict requirement. We could merge this first, based on unit tests and manual testing of the feature. Then separately we could finalize the integration tests. 

OK, I'll update it soon. 

> Have you already manually tested the crypt devices for this PR?

Yes, I've already used the official upstream Rook + this feature's patch for some months and it has worked fine.

**Comment by satoru-takeuchi** [comment]
@travisn @leseb I reflect leseb's comment and rebase this PR to master. PTAL.

File: pkg/clusterd/disk.go [hunk 1]
```diff
@@ -35,7 +35,7 @@ var (
 )
 
 func supportedDeviceType(device string) bool {
-	return device == sys.DiskType || device == sys.SSDType || device == sys.LVMType || device == sys.MultiPath || device == sys.PartType || device == sys.LinearType
+	return device == sys.DiskType || device == sys.SSDType || device == sys.CryptType || device == sys.LVMType || device == sys.MultiPath || device == sys.PartType || device == sys.LinearType
```
[the remark this answers is not in the document]
Comment (satoru-takeuchi) [reply pkg/clusterd/disk.go]: fixed, thanks.

**Pull request merged**